vault/version_store.go

```diff
@@ -70,7 +70,7 @@ func (c *Core) storeVersionTimestamp(ctx context.Context, version string, timest
 // FindOldestVersionTimestamp searches for the vault version with the oldest
 // upgrade timestamp from storage. The earliest version this can be is 1.9.0.
 func (c *Core) FindOldestVersionTimestamp() (string, time.Time, error) {
-	if len(c.versionTimestamps) == 0 {
+	if c.versionTimestamps == nil {
```

Comment:
I think these don't need to be nil checks right? Checking the length for a nil value should result in 0.

Reply:
You're right, but in this case, having a zero length struct is not an error condition. We should only be erroring if the struct is actually nil.

Reply:
Ah, yes that makes sense!